frontend/public/components/storage-class-form.tsx

```diff
@@ -64,105 +77,35 @@ export class StorageClassForm_ extends React.Component<
-        },
-        'csi.storage.k8s.io/provisioner-secret-name': {
-          name: 'Provisioner Secret Name',
-          hintText: 'The name of provisioner secret',
-          required: true,
-        },
-        'csi.storage.k8s.io/provisioner-secret-namespace': {
-          name: 'Provisioner Secret Namespace',
-          hintText: 'The namespace where provisioner secret is created',
-          required: true,
-        },
-        'csi.storage.k8s.io/node-stage-secret-name': {
-          name: 'Node Stage Secret Name',
-          hintText: 'The name of Node Stage secret',
-          required: true,
-        },
-        'csi.storage.k8s.io/node-stage-secret-namespace': {
-          name: 'Node Stage Secret Namespace',
-          hintText: 'The namespace where provisioner secret is created',
-          required: true,
-        },
-        'csi.storage.k8s.io/fstype': {
-          name: 'Filesystem Type',
-          hintText: 'Ceph RBD filesystem type. Default set to ext4',
-          required: true,
-        },
-      },
-    },
-    'cephfs.csi.ceph.com': {
-      title: 'Ceph FS',
-      provisioner: 'cephfs.csi.ceph.com',
-      documentationLink: 'https://rook.io/docs/rook/v1.1/',
-      parameters: {
-        clusterID: {
-          name: 'Cluster ID',
-          hintText: 'The namespace where Ceph is deployed',
-          required: true,
-        },
-        pool: {
-          name: 'Pool',
-          hintText: 'Ceph pool into which volume data shall be stored',
-        },
-        fsName: {
-          name: 'Filesystem Name',
-          hintText: 'CephFS filesystem name into which the volume shall be created',
-          required: true,
-        },
-        'csi.storage.k8s.io/provisioner-secret-name': {
-          name: 'Provisioner Secret Name',
-          hintText: 'The name of provisioner secret',
-          required: true,
-        },
-        'csi.storage.k8s.io/provisioner-secret-namespace': {
-          name: 'Provisioner Secret Namespace',
-          hintText: 'The namespace where provisioner secret is created',
-          required: true,
-        },
-        'csi.storage.k8s.io/node-stage-secret-name': {
-          name: 'Node Stage Secret Name',
-          hintText: 'The name of Node Stage secret',
-          required: true,
-        },
-        'csi.storage.k8s.io/node-stage-secret-namespace': {
-          name: 'Node Stage Secret Namespace',
-          hintText: 'The namespace where provisioner secret is created',
-          required: true,
-        },
+  // Fetch Storage type provisioners from different operators
+  // For CSI - provisionerType: 'csi'
+  // For Defaults - provisionerType: 'others'
+  getExtensionsStorageClassProvisioners = (provisionerType = Provisioner.OTHERS) => {
+    const extensionCSIProvisioners: ExtensionSCProvisionerProp = _.reduce(
+      this.props.params,
+      (res, value) => {
+        const obj = value.properties.getStorageClassProvisioner || {};
+        if (obj) {
```

Comment:
i think you still need to return `res`

```suggestion
        if (obj) {
          const key = provisionerType;
          const keyValue = obj[provisionerType];
          res[key] = keyValue;
        }
        return res;
```